src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -1025,6 +1026,45 @@ class spell_sha_thunderstorm : public SpellScriptLoader
         }
 };
 
+// 18350 - Totemic Mastery (Tier 6 - 2P)
+class spell_sha_totemic_mastery : public SpellScriptLoader
+{
+public:
+    spell_sha_totemic_mastery() : SpellScriptLoader("spell_sha_totemic_mastery") { }
+
+    class spell_sha_totemic_mastery_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_sha_totemic_mastery_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_TOTEMIC_MASTERY))
+                return false;
+            return true;
+        }
+
+        void HandleDummy(SpellEffIndex /*effIndex*/)
+        {
+            Unit* caster = GetCaster();
+            for (uint8 i = SUMMON_SLOT_TOTEM; i < MAX_TOTEM_SLOT; ++i)
+                if (!caster->m_SummonSlot[i])
+                    return;
+
+            caster->CastSpell(caster, SPELL_SHAMAN_TOTEMIC_MASTERY, true);
```

Comment:
The aura lasts 5 seconds when one of the totem is destroyed / removed it is removed by expires. Is it really necessary to remove the aura code in this case?!